addons/product/views/product_views.xml

```diff
@@ -182,6 +182,7 @@
                 <group expand="1" string="Group By">
                     <filter string="Product Type" name="type" context="{'group_by':'type'}"/>
                     <filter string="Product Category" name="categ_id" context="{'group_by':'categ_id'}"/>
+                    <filter string="Product Template" name="name" context="{'group_by':'name'}"/>
```

Comment:
AFAICS the only point that could be related to this change in the spec is marked as dropped :thinking: 
Also, the view you want to change is probably the `product_search_form_view`, not the template one.
This group filter should target variants and not templates if I understand correctly.

Reply:
Good catch! :+1: I have indeed crossed out that requirement as cancelled, but that's because sums cannot be displayed for group views (Odoo framework is not capable of that). However, grouping by project template is completely doable and, IMO, independent from summing up the values.

I have updated the task: now the requirement is no longer crossed, and a note specifies that we're dropping sums only, keeping the grouping by product template.

As for the view, I think this is the right one. The new option is shown as expected, and I can see that another module (`point_of_sale`) modifies the same view to add its "POS Product Category" grouping. Could you please explain why this might be a wrong one?

Reply:
The view you're modifying is the default search view for `product.template`, and the base of the default search view for `product.product` records.  This means that your new filter grouping records by 'Product Template' will also be shown in a `product.template` view, which seems meaningless ?

IIUC, Either the name of your filter is wrong, because you want to group products (variants or templates) by name, or you want to group variants of the same template, either your filter should be in the `product.product` search view, and group by `product_tmpl_id` instead.

Reply:
I've moved the `<filter>` element to a different view, and it now uses `product_tmpl_id`. Is this what you had in mind? :)

Reply:
Exactly :+1: